src/extras.ts

```diff
@@ -89,3 +89,47 @@ export let deleteMyDocumentsAsync = async (storage: IStorage | IStorageAsync, ke
         numErrors,
     };
 }
+
+export let copyMyDocsToOtherWorkspace = async (sourceStorage: IStorage | IStorageAsync, destStorage: IStorage | IStorageAsync, keypair: AuthorKeypair) => {
+    
+    // Get all docs by Author from sourceStorage
+    let myDocs = await sourceStorage.documents({
+        author: keypair.address,
+        history: 'all',
+    });
+
+    storageLogger.log(`copying ${myDocs.length} docs authored by ${keypair.address} from ${sourceStorage.workspace} to ${destStorage.workspace}...`);
+
+    let numErrors = 0;
+    let numCopied = 0;
+
+    // Write them to destStorage.  Modify the workspace property but preserve timestamps.
+    for(let doc of myDocs) {
+        let copiedDoc: DocToSet = {
+            format: doc.format,
+            path: doc.path,
+            content: doc.content,
+            timestamp: doc.timestamp,
+            deleteAfter: doc.deleteAfter
+        };
+
+        let result = await destStorage.set(keypair, copiedDoc);
+
+        if(isErr(result)) {
+            storageLogger.error(`copying ${doc.path}... error`);
+            numErrors += 1;
+        } else if(result === WriteResult.Ignored) {
+            storageLogger.log(`copying ${doc.path}... ignored`);
+            numErrors += 1;
```

Comment:
Even though it'd be strange for this to happen, I'm not sure it should be flagged as an error. Perhaps another property, `numIgnored`, which only displays in the rare event this happens?

Reply:
Agreed.  I'll add `numIgnored` to the return object to indicate the number of ignored writes to the `destStorage`.

Reply:
"Ignored" happens when the document is obsolete, e.g. there's a newer one already present in the destination workspace.

That could definitely happen if you migrate into an existing workspace; it should never happen if you migrate to a new and fresh workspace (if you can do it fast enough with no other writes happening at the same time... :)

Good idea to also count the `numIgnored` and return that in the summary.